File: dart-client/lib/model/user_credentials_object.dart
part of swagger.api;

class UserCredentialsObject {
  
  String username = null;
  

  String password = null;
  
  UserCredentialsObject();

  @override
  String toString() {
    return 'UserCredentialsObject[username=$username, password=$password, ]';
  }

  UserCredentialsObject.fromJson(Map<String, dynamic> json) {
    if (json == null) return;
    username =
        json['username']
    ;
    password =
        json['password']
    ;
  }

  Map<String, dynamic> toJson() {
    return {
      'username': username,
      'password': password
     };
  }

  static List<UserCredentialsObject> listFromJson(List<dynamic> json) {
    return json == null ? new List<UserCredentialsObject>() : json.map((value) => new UserCredentialsObject.fromJson(value)).toList();
  }

  static Map<String, UserCredentialsObject> mapFromJson(Map<String, Map<String, dynamic>> json) {
    var map = new Map<String, UserCredentialsObject>();
    if (json != null && json.length > 0) {
      json.forEach((String key, Map<String, dynamic> value) => map[key] = new UserCredentialsObject.fromJson(value));
    }
    return map;
  }
}

File: tizen-client/src/UserCredentialsObject.cpp
#include <map>
#include <cstdlib>
#include <glib-object.h>
#include <json-glib/json-glib.h>
#include "Helpers.h"


#include "UserCredentialsObject.h"

using namespace std;
using namespace Tizen::ArtikCloud;

UserCredentialsObject::UserCredentialsObject()
{
	//__init();
}

UserCredentialsObject::~UserCredentialsObject()
{
	//__cleanup();
}

void
UserCredentialsObject::__init()
{
	//
	//
	//username = std::string();
	//
	//
	//password = std::string();
	//
}

void
UserCredentialsObject::__cleanup()
{
	//if(username != NULL) {
	//
	//delete username;
	//username = NULL;
	//}
	//if(password != NULL) {
	//
	//delete password;
	//password = NULL;
	//}
	//
}

void
UserCredentialsObject::fromJson(char* jsonStr)
{
	JsonObject *pJsonObject = json_node_get_object(json_from_string(jsonStr,NULL));
	JsonNode *node;
	const gchar *usernameKey = "username";
	node = json_object_get_member(pJsonObject, usernameKey);
	if (node !=NULL) {
	

		if (isprimitive("std::string")) {
			jsonToValue(&username, node, "std::string", "");
		} else {
			
		}
	}
	const gchar *passwordKey = "password";
	node = json_object_get_member(pJsonObject, passwordKey);
	if (node !=NULL) {
	

		if (isprimitive("std::string")) {
			jsonToValue(&password, node, "std::string", "");
		} else {
			
		}
	}
}

UserCredentialsObject::UserCredentialsObject(char* json)
{
	this->fromJson(json);
}

char*
UserCredentialsObject::toJson()
{
	JsonObject *pJsonObject = json_object_new();
	JsonNode *node;
	if (isprimitive("std::string")) {
		std::string obj = getUsername();
		node = converttoJson(&obj, "std::string", "");
	}
	else {
		
	}
	const gchar *usernameKey = "username";
	json_object_set_member(pJsonObject, usernameKey, node);
	if (isprimitive("std::string")) {
		std::string obj = getPassword();
		node = converttoJson(&obj, "std::string", "");
	}
	else {
		
	}
	const gchar *passwordKey = "password";
	json_object_set_member(pJsonObject, passwordKey, node);
	node = json_node_alloc();
	json_node_init(node, JSON_NODE_OBJECT);
	json_node_take_object(node, pJsonObject);
	char * ret = json_to_string(node, false);
	json_node_free(node);
	return ret;
}

std::string
UserCredentialsObject::getUsername()
{
	return username;
}

void
UserCredentialsObject::setUsername(std::string  username)
{
	this->username = username;
}

std::string
UserCredentialsObject::getPassword()
{
	return password;
}

void
UserCredentialsObject::setPassword(std::string  password)
{
	this->password = password;
}


